notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationService.java

```diff
@@ -94,7 +95,26 @@ public NotificationResponse fetch(PortletRequest req) {
         return rslt;
 
     }
+	
+    public void updateEntryState(PortletRequest req, String entryId, NotificationState state) {
```

Comment:
Might want to call this call this addNotificationState.  Should probably specify in docs that caller must insure the state does not already exist to avoid creating multiple of the same state value (there are currently no restrictions on that which we may want to change in a separate PR).  In the future this method signature might need to go into the interface also but for now I think it is OK it is only in the service.